### PR TITLE
replace ant tasks kubectl usage by uyunictl

### PR DESCRIPTION
**Pull request by rjpmestre**
## What does this PR change?

replace ant tasks kubectl usage by uyunictl

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests

- [x] **DONE**

## Links

Fixes #
None

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/manager-build.xml [hunk 1]
```diff
@@ -17,6 +17,10 @@
   <property file="conf/rhn_java.conf"/>
 
   <!-- Other properties -->
+  <property name="container.name" value="uyuni-server" />
```
Comment (cbosdo) [inline java/manager-build.xml:20]: You can remove the `container.name` property: this is automatically discovered by `uyunictl`

File: java/manager-build.xml [hunk 2]
```diff
@@ -17,6 +17,10 @@
...
+  <property name="ssh.key" value="/home/user/.ssh/id_rsa" />
+  <property name="podman.connection.name" value="srv" />
```
Comment (cbosdo) [inline java/manager-build.xml:22]: I would let the user configure their connections at the `podman` level and only pass the connection name to the ant target. What do you guys think about it? In any case for kubernetes users they would need to setup they kubeconfig file to connect to the cluster.

File: java/manager-build.xml [hunk 3]
```diff
@@ -298,135 +302,156 @@
...
+    <fail message="The specified ssh.key file does not exist">
+      <condition>
+        <not>
+          <available file="${ssh.key}"/>
+        </not>
+      </condition>
+    </fail>
+  </target>
+
+  <target name="is-podman-installed" description="Cheks if podman is installed">
```
Comment (cbosdo) [inline java/manager-build.xml:330]: We don't care if `podman` or `kubectl` is installed: `uyunictl` checks for us, and even checks for `podman-remote`.

File: java/manager-build.xml [hunk 4]
```diff
@@ -298,135 +302,156 @@
...
-      <arg line="exec ${kubectl.args} -- mv ${deploy.dir}/WEB-INF/lib/rhn.jar /usr/share/rhn/lib"/>
-    </exec>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf /usr/share/rhn/lib/rhn.jar ${deploy.dir}/WEB-INF/lib"/>
+  </target>
+
+  <target name="remove-podman-connection" depends="warn-if-podman-not-installed" description="Removes podman system connection">
+    <echo message="Removing podman system connection..."/>
+    <exec executable="podman">
+      <arg line="system connection rm ${podman.connection.name}"/>
```
Comment (cbosdo) [inline java/manager-build.xml:362]: I probably mislead you here, but the more I think about it, the more I'm convinced we don't want any `podman` or `kubectl` specific code here. We can just document how to setup the connections. Setting up a remote connection on `podman` doesn't prevent from using the local `podman` instance if the user doesn't set the `CONTAINER_CONNECTION` variable or the `--connection` parameter.

File: java/manager-build.xml [hunk 5]
```diff
@@ -298,135 +302,156 @@
...
   </target>
 
-  <target name="deploy-static-resources-kube" depends="build-js,warn-if-kubectl-not-installed" description="Deploy css/js/img/font files in a running kubernetes pod">
-    <echo message="Copying frontend files to remote host..."/>
+  <target name="deploy-container" depends="webapp, add-podman-connection" description="Deploy a new copy of SUSE Manager in a running pod container">
+    <echo message="Copying files to pod container... ${build.dir} -> ${deploy.dir}" />
+
     <exec failonerror="true" executable="sh">
-        <arg line="-c 'tar c -p -C ${frontend.dist.dir} -f - . | kubectl exec -i ${kubectl.args} -- tar xf - -C /srv/www/htdocs'"/>
+      <arg line="-c 'tar c -C ${build.dir}/webapp -f - --exclude=log4j2.xml . | uyunictl exec -i -- tar xf - -C ${deploy.dir}'"/>
```
Comment (cbosdo) [inline java/manager-build.xml:370]: I don't remember why I had to use `tar` here, but it seems that `kubectl` can copy folders too. In any case this should be handled at the `uyunictl` level, not the ant file to make `uyunictl` user's life easier. Is there any reason not to use your fancy `copy-to-container` macro?

File: java/manager-build.xml [hunk 6]
```diff
@@ -298,135 +302,156 @@
...
-        <arg value="-c"/>
-        <arg value="echo 'JAVA_OPTS=&quot;-Xdebug -Xrunjdwp:transport=dt_socket,address=\$HOSTNAME:8000,server=y,suspend=n&quot;' >>/etc/sysconfig/tomcat"/>
-    </exec>
-  </target>
 
-  <target name="replace-tomcat-debug-kube" depends="check-tomcat-has-opts" if="tomcat.has_opts">
-    <exec executable="kubectl">
-        <arg line="exec ${kubectl.args} -- sed 's/JAVA_OPTS=&quot;\([^&quot;]*\)&quot;/JAVA_OPTS=&quot;-Xdebug -Xrunjdwp:transport=dt_socket,address=\$HOSTNAME:8000,server=y,suspend=n \1&quot;/' -i /etc/sysconfig/tomcat"/>
+  <!-- wraps uyunictl command, executing the given argline and setting the CONTAINER_CONNECTION env var -->
+  <macrodef name="wrap-uyunictl">
```
Comment (cbosdo) [inline java/manager-build.xml:430]: I love the way you implemented this with a macro 😍 . May be you can name it simply `uyunictl`

File: java/manager-build.xml [hunk 7]
```diff
@@ -298,135 +302,156 @@
...
+    <attribute name="argline"/>
+    <attribute name="failonerror" default="true"/>
+    <sequential>
+      <exec failonerror="@{failonerror}" executable="uyunictl">
+        <arg line="@{argline}"/>
+        <env key="CONTAINER_CONNECTION" value="${podman.connection.name}"/>
```
Comment (cbosdo) [inline java/manager-build.xml:436]: This variable is not useful in the case of `kubectl` but can't harm if it's added unconditionnaly

File: java/manager-build.xml [hunk 8]
```diff
@@ -298,135 +302,156 @@
...
-    <exec executable="kubectl">
-      <arg line="exec ${kubectl.args} -- sed 's/JAVA_OPTS=&quot;&quot;/JAVA_OPTS=&quot;-Xdebug -Xrunjdwp:transport=dt_socket,address=\$HOSTNAME:8001,server=y,suspend=n&quot;/' -i /etc/rhn/taskomatic.conf"/>
+  <!-- copies the given source dir to the given target dir in the container -->
+  <macrodef name="copy-to-container">
+    <attribute name="source"/>
+    <attribute name="target"/>
+    <attribute name="failonerror" default="false"/>
+    <sequential>
+      <exec failonerror="@{failonerror}" executable="sh">
+        <arg line="-c 'tar c -p -C @{source} -f - . | uyunictl exec -i -- tar xf - -C @{target}'"/>
```
Comment (cbosdo) [inline java/manager-build.xml:448]: Probably we can get rid of the `tar` part here.

**Review by cbosdo — COMMENTED** [review]
I probably provided different instructions in the beginning, the PR looks good, but I would remove the `podman` specific parts as mentioned in the comments

File: java/manager-build.xml (hunk 5 above)
Comment (rjpmestre) [reply java/manager-build.xml:370]: As we agreed, on this specific case we're sticking to the tar solution. The reason is that kubectl doesn't support excluding specific files/folder during the copy operation.

**Review by cbosdo — APPROVED** [review]
Now you just need to rebase and merge! Thanks for the PR

**Pull request merged**